experimental/dds/tree2/src/feature-libraries/sequence-field/moveEffectTable.ts

```diff
@@ -195,7 +305,137 @@ export function applyMoveEffectsToMark<T>(
 	consumeEffect: boolean,
 	composeChildren?: (a: T | undefined, b: T | undefined) => T | undefined,
 ): Mark<T>[] {
-	if (isMoveMark(mark)) {
+	if (isTransientEffect(mark)) {
+		if (isMoveDestination(mark.attach)) {
+			if (isMoveSource(mark.detach)) {
+				return [mark];
+			}
+			const attachRevision = mark.attach.revision ?? mark.revision ?? revision;
+			const effect = getMoveEffect(
+				effects,
+				CrossFieldTarget.Destination,
+				attachRevision,
+				mark.attach.id,
+				mark.count,
+			);
+
+			if (effect.length < mark.count) {
+				const [firstMark, secondMark] = splitMark(mark, effect.length);
+				const updatedAttach = { ...(firstMark.attach as MoveDestination) };
+				applyMoveEffectsToDest(
```

Comment:
Why the copy and not just:
```suggestion
				const updatedAttach = firstMark.attach as MoveDestination;
```